hardware/evohome.cpp

```diff
@@ -173,15 +178,42 @@ bool CEvohome::StartHardware()
 		m_retrycntr=RETRY_DELAY; //will force reconnect first thing
 		
 		std::vector<std::vector<std::string> > result;
-		result = m_sql.safe_query("SELECT Name,DeviceID,nValue FROM DeviceStatus WHERE (HardwareID==%d) AND (Unit==0)", m_HwdID);
+		result = m_sql.safe_query("SELECT Name,DeviceID,nValue FROM DeviceStatus WHERE (HardwareID==%d) AND (Unit==0) AND (Type==%d)", m_HwdID, (int)pTypeEvohome);
+			
```

Comment:
I've added the additional Type==pTypeEvohome check here as I'm identifying the Device ID of the controller.  I've discovered through testing that other Evohome devices can be created with Unit==0 (relays and zone temeperature sensors).  Without this additional check, this query can return multiple devices leading to mis-identification of the Controller ID.